contract_sale_generation/models/contract_line.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (C) 2020 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ContractLine(models.Model):
+    _inherit = "contract.line"
+
+    def _prepare_sale_line_vals(self, dates, order_id=False):
+        sale_line_vals = {
+            "product_id": self.product_id.id,
+            "product_uom_qty": self._get_quantity_to_invoice(*dates),
+            "product_uom": self.uom_id.id,
+            "discount": self.discount,
+            "contract_line_id": self.id,
+            "display_type": self.display_type,
+        }
+        if order_id:
+            sale_line_vals["order_id"] = order_id.id
+        return sale_line_vals
+
+    def _prepare_sale_line(self, order_id=False, sale_values=False):
+        self.ensure_one()
+        dates = self._get_period_to_invoice(
+            self.last_date_invoiced, self.recurring_next_date
+        )
+        sale_line_vals = self._prepare_sale_line_vals(dates, order_id)
```

Comment:
Call keyword parm with keyword: `order_id=order_id`

Reply:
Actually `order_id` is nowhere passed to `_prepare_sale_line` as far as I can see. So logic and argument can maybe removed from this function and from `_prepare_sale_line_vals`. Can anyoint point out a use for thus argument?

Reply:
I think this is intended in order to override and force an existing order